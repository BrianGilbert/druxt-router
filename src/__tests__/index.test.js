import druxtRouter from '..'

// const baseURL = 'https://example.com'
const baseURL = 'http://contenta.druxt.localhost'

describe('Druxt router', () => {
  test('constructor', () => {
    // Throw error if 'baseURL' not provided.
    expect(() => { new druxtRouter }).toThrow('The \'baseURL\' parameter is required.')

    // Ensure class type.
    expect(new druxtRouter(baseURL)).toBeInstanceOf(druxtRouter)
  })

  test('get', async () => {
    const router = new druxtRouter(baseURL)
    const result = await router.get('/')

    expect(result.entity).toBeDefined()
    expect(result.isHomePath).toBeTruthy()
    expect(result.jsonapi).toBeDefined()
    expect(result.label).toBeDefined()
    expect(result.resolved).toBeDefined()
    expect(result).toMatchSnapshot()
  })
})
