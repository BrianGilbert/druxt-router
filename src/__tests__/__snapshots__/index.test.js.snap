// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Druxt router get 1`] = `
Object {
  "entity": Object {
    "bundle": "page",
    "canonical": "http://contenta.druxt.localhost/welcome",
    "id": "6",
    "type": "node",
    "uuid": "5fd39fb3-5d2f-4f9a-a50a-696df984a6db",
  },
  "isHomePath": true,
  "jsonapi": Object {
    "basePath": "/api",
    "entryPoint": "http://contenta.druxt.localhost/api",
    "individual": "http://contenta.druxt.localhost/api/pages/5fd39fb3-5d2f-4f9a-a50a-696df984a6db",
    "pathPrefix": "api",
    "resourceName": "pages",
  },
  "label": "Welcome to Contenta CMS!",
  "meta": Object {
    "deprecated": Object {
      "jsonapi.pathPrefix": "This property has been deprecated and will be removed in the next version of Decoupled Router. Use basePath instead.",
    },
  },
  "resolved": "http://contenta.druxt.localhost/welcome",
}
`;
